src/libasr/asr_scopes.cpp

```diff
@@ -135,4 +135,104 @@ std::string SymbolTable::get_unique_name(const std::string &name) {
+                        char *es_name = ASR::down_cast<
+                            ASR::ExternalSymbol_t>(s)->m_module_name;
+                        if (!present(mod_dependencies, es_name)) {
+                            mod_dependencies.push_back(al, es_name);
+                        }
+                    }
+                }
+                fn->m_symtab->parent = module_scope;
+                module_scope->add_symbol(a.first, (ASR::symbol_t *) fn);
+                syms.push_back(al, s2c(al, a.first));
+                break;
+            } case (ASR::symbolType::GenericProcedure) : {
+                ASR::GenericProcedure_t *es = ASR::down_cast<ASR::GenericProcedure_t>(a.second);
+                es->m_parent_symtab = module_scope;
+                module_scope->add_symbol(a.first, (ASR::symbol_t *) es);
+                syms.push_back(al, s2c(al, a.first));
+                break;
+            } case (ASR::symbolType::ExternalSymbol) : {
+                ASR::ExternalSymbol_t *es = ASR::down_cast<ASR::ExternalSymbol_t>(a.second);
+                if (!present(mod_dependencies, es->m_module_name)) {
+                    mod_dependencies.push_back(al, es->m_module_name);
+                }
+                es->m_parent_symtab = module_scope;
+                module_scope->add_symbol(a.first, (ASR::symbol_t *) es);
+                syms.push_back(al, s2c(al, a.first));
+                break;
+            } case (ASR::symbolType::StructType) : {
+                ASR::StructType_t *st = ASR::down_cast<ASR::StructType_t>(a.second);
+                st->m_symtab->parent = module_scope;
+                module_scope->add_symbol(a.first, (ASR::symbol_t *) st);
+                syms.push_back(al, s2c(al, a.first));
+                break;
+            } case (ASR::symbolType::EnumType) : {
+                ASR::EnumType_t *et = ASR::down_cast<ASR::EnumType_t>(a.second);
+                et->m_symtab->parent = module_scope;
+                module_scope->add_symbol(a.first, (ASR::symbol_t *) et);
+                syms.push_back(al, s2c(al, a.first));
+                break;
+            } case (ASR::symbolType::UnionType) : {
+                ASR::UnionType_t *ut = ASR::down_cast<ASR::UnionType_t>(a.second);
+                ut->m_symtab->parent = module_scope;
+                module_scope->add_symbol(a.first, (ASR::symbol_t *) ut);
+                syms.push_back(al, s2c(al, a.first));
+                break;
+            } case (ASR::symbolType::Variable) : {
+                ASR::Variable_t *v = ASR::down_cast<ASR::Variable_t>(a.second);
+                v->m_parent_symtab = module_scope;
+                module_scope->add_symbol(a.first, (ASR::symbol_t *) v);
+                syms.push_back(al, s2c(al, a.first));
+                break;
+            } default : {
+                throw LCompilersException("Moving the symbol:`" + a.first +
+                    "` from global scope is not implemented yet");
+            };
+        }
```

Comment:
This isn't scalable. I would do it by writing a visitor in `asdl_cpp.py`. Will you be able to try the class design and implement it? Give it a try for a couple of days, if not then we will merge this with a `TODO`.

Reply:
Added this in TODO